### PR TITLE
fix(Overlay): prevent double-calling onRequestClose

### PR DESCRIPTION
**Pull request by jakemhiller**
### PR Checklist

- [ ] Related to Abstract designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut/src/Modal/__tests__/Modal-test.tsx [hunk 1]
```diff
@@ -63,6 +63,16 @@ describe('Modal', () => {
     expect(onRequestClose.mock.calls.length).toBe(1);
   });
 
+  it('triggers onRequestClose callback when clicking the default close button', () => {
```
Comment (hmafzal) [inline packages/gamut/src/Modal/__tests__/Modal-test.tsx:66]: Shouldn't there also be a test to verify that the onRequestClose callback only fires once when clicking the non-default close button (eg, a button within modal)?

File: packages/gamut/src/Overlay/index.tsx [hunk 2]
```diff
@@ -32,7 +43,29 @@ export const Overlay: React.FC<OverlayProps> = ({
...
+    if (!wasOpen && isOpen && !trapActive) {
+      setTrapActive(true);
+    }
+    // Parent component set isOpen to false
+    if (wasOpen && !isOpen && trapActive) {
+      setTrapActive(false);
+    }
+  }, [isOpen, onRequestClose, trapActive, wasOpen]);
+
+  if (!trapActive) return null;
```
Comment (hmafzal) [inline packages/gamut/src/Overlay/index.tsx:68]: A little nervous about all this logic around whether it's open or not.. but I also can't see getting around something like this if FocusTraps onDeactivate is called when FocusTrap unmounts.

File: packages/gamut/src/Overlay/index.tsx [hunk 3]
```diff
@@ -32,7 +43,29 @@ export const Overlay: React.FC<OverlayProps> = ({
   onRequestClose,
   isOpen,
 }) => {
-  if (!isOpen) return null;
+  const [trapActive, setTrapActive] = useState(isOpen);
+  const wasOpen = usePrevious(isOpen);
+
+  const requestCloseCallback = () => {
```
Comment (hmafzal) [inline packages/gamut/src/Overlay/index.tsx:49]: nit: rename to `onDeactivate`/`onDeactivateFocusTrap` to make distinction from request close

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.9.5-alpha.b076b4.0<br/>@codecademy/gamut@12.3.5-alpha.b076b4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f36eddc19cb860092115af1--gamut-preview.netlify.app>https://5f36eddc19cb860092115af1--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f36eddc19cb860092115af1>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.9.5-alpha.a8b946.0<br/>@codecademy/gamut@12.3.5-alpha.a8b946.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f36eeccb1ad9e00bb57262d--gamut-preview.netlify.app>https://5f36eeccb1ad9e00bb57262d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f36eeccb1ad9e00bb57262d>Deploy Logs</a></p>

File: packages/gamut/src/Overlay/index.tsx (hunk 2 above)
Comment (jakemhiller) [reply packages/gamut/src/Overlay/index.tsx:60]: Agreed.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.9.6-alpha.feaf9a.0<br/>@codecademy/gamut@13.0.1-alpha.feaf9a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f3c2974168ad7d1331e406c--gamut-preview.netlify.app>https://5f3c2974168ad7d1331e406c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f3c2974168ad7d1331e406c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.9.6-alpha.f4c1f6.0<br/>@codecademy/gamut@13.0.1-alpha.f4c1f6.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@1.9.6-alpha.9c58d8.0<br/>@codecademy/gamut@13.0.1-alpha.9c58d8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f3c51ed0a2d52009a9a7ff8--gamut-preview.netlify.app>https://5f3c51ed0a2d52009a9a7ff8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f3c51ed0a2d52009a9a7ff8>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f3c51f59b26570118307e0b--gamut-preview.netlify.app>https://5f3c51f59b26570118307e0b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f3c51f59b26570118307e0b>Deploy Logs</a></p>

**Pull request closed without merging**